lib/scope.go

```diff
@@ -20,9 +22,46 @@ import (
 type scope struct {
 	ctx  context.Context
 	inst *Instance
+	pro  *profile.Profile
 	// TODO(dustmop): Additional information, such as user identity, their profile, keys
 }
 
+func newScope(ctx context.Context, inst *Instance) (scope, error) {
+	var pro *profile.Profile
+	// TODO(b5): shouldn't need this nil checking
+	if inst != nil && inst.profiles != nil {
+		pro = inst.profiles.Owner()
+		if tokenString := token.FromCtx(ctx); tokenString != "" {
+			tok, err := token.ParseAuthToken(tokenString, inst.keystore)
+			if err != nil {
+				return scope{}, err
+			}
+
+			if claims, ok := tok.Claims.(*token.Claims); ok {
+				// TODO(b5): at this point we have a valid signature of a profileID string
+				// but no proof that this profile is owned by the key that signed the
+				// token. We either need ProfileID == KeyID, or we need a UCAN. we need to
+				// check for those, ideally in a method within the profile package that
+				// abstracts over profile & key agreement
+				pro, err = inst.profiles.GetProfile(profile.IDB58DecodeOrEmpty(claims.ProfileID))
```

Comment:
👍 UCAN would be great as a longer term expansion but think we can get by with ProfileID == KeyID for the time being, given that the keys need to be added from the owner side similar to a whitelisting process.

Reply:
I'll test out the qri-logbook-integration tests and see how this interacts with `prove` changing the profileID. Last I left off cloud was *not* sending back UCAN, but that may be necessary to keep this sane.